### ISSUE TITLE
Markdown: Fenced block in list breaks ignoring spelling of LaTeX/TeX-style commands

### ISSUE THREAD

**Issue by 0112358437189887641562819** [bartosz-antosik/vscode-spellright#128]
I'm using Markdown to write documents that include LaTeX/TeX math blocks. *Spell Right* successfully ignores LaTeX/TeX-style commands ... until you use a fenced block in a list. This doesn't seem to break simple ``ignoreRegExps`` and ``ignoreRegExpsByClass`` settings. The short document below should duplicate this behavior.

Example settings:

``````javascript
// a simple user-defined rule
"spellright.ignoreRegExps": [ "/fabloozle/ig" ]
``````

Example document:

``````markdown
# This document breaks *Spell Right*

Fabloozle is my made-up word. Here's a fenced math block. *Spell Right* works.

```math
\dfrac{2}{4} \Rightarrow \dfrac{1}{2}
```

Here's a list with a fenced block:

-   Fabloozle. Here's the same math, but inline.
    *Spell Right* still works: $`\dfrac{2}{4} \Rightarrow \dfrac{1}{2}`$.

-   Here's the same math, again as a fenced block.
    Now *Spell Right* does *not* work for ``\foo``-style commands:

    ```math
    \dfrac{2}{4} \Rightarrow \dfrac{1}{2}
    ```
    <!-- leaving this fenced block empty also breaks *Spell Right* -->

Yet again, here's the same fenced math block.
*Spell Right* still does *not* work for ``\foo``-style commands.

```math
\dfrac{2}{4} \Rightarrow \dfrac{1}{2}
```

Fabloozle. *Spell Right* still seems to work for simple user-defined
``spellright.ignoreRegExps`` and ``spellright.ignoreRegExpsByClass`` settings.
``````

**Comment by 0112358437189887641562819** [comment]
Using the following rule works around this problem for now.

``````javascript
"spellright.ignoreRegExpsByClass": {
  "markdown": [ "/(\\\\\\[a-z]+)/ig" ]
``````

**Comment by bartosz-antosik** [comment]
Thanks for committing! I have not yet looked into this one.

**Comment by bartosz-antosik** [comment]
This should be corrected in version 2.1.34. Please have a look and tell me whether it does.

My resolution, as few other earlier is a heuristics that should work the case.

Markdown is very tricky though. It would call for a proper parser but there is so many varieties of Markdown out there that it is nearly impossible to do anything about this. Even VSCode itself is sometimes inconsistent about Markdown, as you can see e.g. [here](https://github.com/bartosz-antosik/vscode-spellright/issues/105#issuecomment-354291439).

**Comment by toenails6** [comment]
I recently had this issue, the solution from the following post seemed to have worked: 
https://stackoverflow.com/questions/63424928/how-can-i-disable-spellchecking-of-latex-math-fragments-in-markdown-in-visual-st
```
"spellright.ignoreRegExpsByClass": {
  "markdown": [ 
    "/\\\\\\\\\\([^]*?\\\\\\\\\\)/gm" ,   
    "/\\\\\\\\\\[[^]*?\\\\\\\\\\]/gm" ,
    "/\\$\\$[^]*?\\$\\$/gm" ],
  ]
}
```
